### PR TITLE
Refactor server

### PR DESCRIPTION
**Pull request by palakbhasin**
Added all methods from wishlists.py to server.py

### CODE REVIEW

**Review by pinocchiochang — APPROVED** [review]
Just reformat? OK. Looks good.

File: app/server.py [hunk 1]
```diff
@@ -19,108 +20,269 @@
...
+		item = wl.find_item(item_id)
+		return make_response(jsonify(item), status.HTTP_200_OK)
+	except ItemException:
+		return make_response(jsonify(message='Item with id %s could not be found' % item_id), status.HTTP_404_NOT_FOUND)
+	except WishlistException:
+		return make_response(jsonify(message='Wishlist with id %d could not be found' % wishlist_id), status.HTTP_404_NOT_FOUND)
+
+
+@app.route('/wishlists/<int:id>', methods=['PUT'])
+def update_wishlist(id):
```
Comment (pinocchiochang) [inline app/server.py:134]: When update a wishlist, maybe just update its name or user_id, the items in this wishlist gone.

File: app/server.py [hunk 2]
```diff
@@ -19,108 +20,269 @@
...
+	try:
+		wl = Wishlist.find_or_404(wishlist_id)
+		wl.delete()
+		return make_response('', status.HTTP_204_NO_CONTENT)
+	except WishlistException:
+		return make_response(jsonify(message='Wishlist with id %d could not be found' % wishlist_id), status.HTTP_204_NO_CONTENT)
+
+
+@app.route('/wishlists/search', methods=['GET'])
+def search_wishlists():
```
Comment (pinocchiochang) [inline app/server.py:240]: How is the url of search actuall? I try both 'curl http://0.0.0.0.:5000/wishlists/search?q=new_item' and 'curl http://0.0.0.0.:5000/wishlists/search#q=new_item', or 'curl http://0.0.0.0.:5000/wishlists/search?user_id=123'. It just return no wishlist.

**Review by pinocchiochang — CHANGES_REQUESTED** [review]

File: app/server.py (hunk 2 above)
Comment (palakbhasin) [reply app/server.py:240]: http://0.0.0.0:5000/wishlists/search?q=Apple&user_id=123

File: app/server.py (hunk 1 above)
Comment (palakbhasin) [reply app/server.py:134]: Ok, fixed. Try it again

Comment (pinocchiochang) [reply app/server.py:134]: Ok. works

**Comment by xynazog** [comment]
Great job @palakbhasin 🎉

**Pull request merged**